### PR TITLE
Fix README TOC links

### PR DESCRIPTION
**Pull request by crystalcommunication**
Fixes anchored markup links by adding `user-content-` (which is prepended to IDs)

Closes https://codeberg.org/Codeberg/Community/issues/894

### CODE REVIEW

**Review by yardenshoham — APPROVED** [review]

File: modules/markup/html.go [hunk 1]
```diff
@@ -359,6 +359,10 @@ func visitNode(ctx *RenderContext, procs, textProcs []processor, node *html.Node
 		if attr.Key == "id" && !(strings.HasPrefix(attr.Val, "user-content-") || blackfridayExtRegex.MatchString(attr.Val)) {
 			node.Attr[idx].Val = "user-content-" + attr.Val
 		}
+		// Add user-content- to # links too
+		if attr.Key == "href" && strings.HasPrefix(attr.Val, "#") && !(strings.HasPrefix(attr.Val, "#user-content-") || blackfridayExtRegex.MatchString(strings.TrimPrefix(attr.Val, "#"))) {
```
Comment (delvh) [inline modules/markup/html.go]: Hmm… I'm wondering if we should extract everything after the first `&&` until the `{` into a separate method.

Comment (crystalcommunication) [reply modules/markup/html.go]: This seemed like the least obtrusive way to do it, but I agree that this line is long and difficult to read.

Comment (oliverpool) [reply modules/markup/html.go]: Refactoring suggestion (untested):

```js
for idx, attr := range node.Attr {
	const val = strings.TrimPrefix(attr.Val, "#");
	const shouldHavePrefix = !(strings.HasPrefix(val, "user-content-") || blackfridayExtRegex.MatchString(val));

	if attr.Key == "id" && shouldHavePrefix) {
		node.Attr[idx].Val = "user-content-" + attr.Val
	}
	// Add user-content- to # links too
	if attr.Key == "href" && strings.HasPrefix(attr.Val, "#") && shouldHavePrefix) {
		node.Attr[idx].Val = "#user-content-" + val
	}

	if attr.Key == "class" && attr.Val == "emoji" {
		textProcs = nil
	}
}

```

Comment (crystalcommunication) [reply modules/markup/html.go]: The code above doesn't quite work but with a bit of adjustment it seems good:
```golang
for idx, attr := range node.Attr {
	var val = strings.TrimPrefix(attr.Val, "#");
	var notHasPrefix = !(strings.HasPrefix(val, "user-content-") || blackfridayExtRegex.MatchString(val));

	if attr.Key == "id" && notHasPrefix {
		node.Attr[idx].Val = "user-content-" + attr.Val
	}
	// Add user-content- to # links too
	if attr.Key == "href" && strings.HasPrefix(attr.Val, "#") && notHasPrefix {
		node.Attr[idx].Val = "#user-content-" + val
	}

	if attr.Key == "class" && attr.Val == "emoji" {
		textProcs = nil
	}
}
```
I also renamed the `shouldHavePrefix` variable to `notHasPrefix` just to make it a little more obvious what this does if you're reading the code (the value should be true if the attribute in question does not already have the necessary prefix). The logic could also be flipped here but `hasPrefix` maybe confusing with references to `strings.HasPrefix` in close proximity.

**Comment by zeripath** [comment]
> Fixes anchored markup links by adding `user-content-` (which is prepended to IDs)

But I thought we had JS that would do the search for this and ensure that things would all work out?

**Comment by zeripath** [comment]
in anchors.js

**Comment by fnetX** [comment]
> But I thought we had JS [...] in anchors.js

But why fix it in JS if we can have a fix in the HTML code that also works for environments without Javascript to fix the broken page?

File: modules/markup/html.go [hunk 2]
```diff
@@ -356,9 +356,16 @@ func postProcess(ctx *RenderContext, procs []processor, input io.Reader, output
 func visitNode(ctx *RenderContext, procs, textProcs []processor, node *html.Node) {
 	// Add user-content- to IDs if they don't already have them
```
Comment (delvh) [inline modules/markup/html.go]:
```suggestion
	// Add user-content- to IDs and "#" links if they don't already have them
```

**Review by delvh — APPROVED** [review]

**Comment by lunny** [comment]
make LG-TM work

**Comment by lunny** [comment]
Please send backport

**Pull request merged**